python/src/aiconfig/default_parsers/dalle.py

```diff
@@ -0,0 +1,204 @@
+        prompt = Prompt(
+            name=prompt_name,
+            input=prompt_input,
+            metadata=PromptMetadata(model=model_metadata, parameters=parameters, **kwargs),
+        )
+        return [prompt]
+
+    # TODO (rossdanlm): Update documentation for args
+    async def deserialize(
+        self, prompt: Prompt, aiconfig: "AIConfigRuntime", _options, params: Optional[Dict] = {}
+    ) -> Dict:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            Update this documentation... serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        # Get inputs from aiconfig
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        model_settings = self.get_model_settings(prompt, aiconfig)
+
+        # Build Completion data
+        completion_data = refine_image_completion_params(model_settings)
+        completion_data["prompt"] = resolved_prompt
+        return completion_data
+
+    async def run_inference(self, prompt: Prompt, aiconfig, options, parameters) -> Output:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        # If needed, certify the API key and initialize the OpenAI client
+        if not openai.api_key:
+            openai.api_key = get_api_key_from_environment("OPENAI_API_KEY")
+        if not self.client:
+            self.client = OpenAI(api_key=openai.api_key)
+
+        completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
+
+        # if stream enabled in runtime options and config, then stream. Otherwise don't stream.
+        stream = (options.stream if options else False) and (
+            not "stream" in completion_data or completion_data.get("stream") != False
+        )
+
+        print("Calling image generation. This can take several seconds, please hold on...")
+        response : ImagesResponse = self.client.images.generate(**completion_data)
+
+        outputs = []
+        if not stream:
+            # ImageResponse object also contains a "created" field for timestamp, should I store that somewhere?
+            # Ex: response=ImagesResponse(created=1700347843, data=[...])
+            for execution_count, image_data in enumerate(response.data):
+                output = construct_regular_output(image_data, execution_count)
+                outputs.append(output)
+        else:
+            # TODO (rossdamlm): Enable streaming next diff
```

Comment:
does the dall-e api support streaming? I checked the OpenAI Api source code and I'm not sure I see streaming as an option for image generation

Reply:
I think you're right. Added https://github.com/lastmile-ai/aiconfig/pull/290 to ignore streaming options